lib/taskana-core/src/main/java/pro/taskana/security/CurrentUserContext.java

```diff
@@ -77,6 +79,31 @@ public static String getUserid() {
     return accessIds;
   }
 
+  /**
+   * This method is supposed to skip further permission checks if we are already in a secured
+   * environment. With great power comes great responsibility.
+   *
+   * @param supplier will be executed with admin privileges
+   * @param <T> defined with the supplier return value
+   * @return output from supplier
+   */
+  public static <T> T runAsAdmin(Supplier<T> supplier) {
+
+    Subject subject = Subject.getSubject(AccessController.getContext());
+    if (subject == null) {
+      // dont add authorisation if none is available.
+      return supplier.get();
+    }
+    Set<Principal> principals = subject.getPrincipals();
+    Set<Object> privateCredentials = subject.getPrivateCredentials();
+    Set<Object> publicCredentials = subject.getPublicCredentials();
+
+    principals.add(new GroupPrincipal("admin"));
```

Comment:
This only works if you do have a user called "admin" configured as admin in taskana.properties, doesn't it? I guess we have to think about this a bit more.

Reply:
Don't we change the current Subject as well, if we do it this way? From my perspective it might be better to create a dedicated Subject from scratch instead of patching the current subject.

Reply:
we could reuse method JobScheduler.getAdminSubject() that creates an admin subject.
May be, the method should be factored out, e.g. moved to InternalTaskanaEngine ...

Reply:
I fixed that issue and added a test to verify that this works as intended